addons/im_livechat/static/src/embed/common/suggestion_service_patch.js

```diff
@@ -0,0 +1,17 @@
+import { SuggestionService } from "@mail/core/common/suggestion_service";
+
+import { patch } from "@web/core/utils/patch";
+
+patch(SuggestionService.prototype, {
+    /** @override */
+    getSupportedDelimiters(thread) {
+        const res = super.getSupportedDelimiters(...arguments);
+        if (thread.channel_type === "livechat") {
+            return res.filter(
+                (delimiter) =>
+                    this.store.self.type !== "guest" || !["#", "/", "::"].includes(delimiter.at(0))
+            );
+        }
+        return res;
+    },
```

Comment:
I'm not fond of such patches, especially since the issue isn't directly related to live chat channels. For example, using a command on a public page as a guest results in a `SessionExpired` error. Instead, we should provide a way to determine whether a command is relevant in the current context and display it only if applicable. Perhaps we could allow commands to register a predicate in the registry.